### ISSUE TITLE
Write specific functions to collapse/rebuild input/outputs as `Raster`, `np.ma.masked_array` or `np.ndarray` consistently

### ISSUE THREAD

**Issue by rhugonnet** [GlacioHack/xdem#293]
We have tons of "wrapper" functions that can take either `Raster`, `np.ma.masked_array` or `np.ndarray` as input, but then only work on the `np.ndarray` (and possibly some metadata), and for output rebuild a `Raster`, `np.ma.masked_array` or `np.ndarray` to match the input.
Right now this is done manually everywhere, we need to have consistent input/output functions for those conversions. Might be triggering issues in #291 due to https://github.com/GlacioHack/GeoUtils/pull/265.

**Comment by erikmannerfelt** [comment]
That's a good idea. A small problem right now (or an unintended feature) is that all metadata are lost after raster recreation. Only the CRS and transform are identical to the input.

**Comment by rhugonnet** [comment]
Yes, the 10 lines in `Coreg.apply()` for "reconstructing" for example: https://github.com/GlacioHack/xdem/blob/main/xdem/coreg/base.py#L1160

**Comment by rhugonnet** [comment]
We could do the same to construct masks from a `Vector`, a `Mask`, an array, etc... It is used everywhere!
Something like this: https://github.com/GlacioHack/xdem/blob/main/xdem/spatialstats.py#L593
